clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -51,7 +54,13 @@
     private final Selectable selector;
     
     private final MetadataUpdater metadataUpdater;
-
+    
+    /* a list of nodes we've connected to in the past */
+    private final List<Integer> nodesEverSeen;
+    private final Map<Integer, Node> nodesEverSeenById;
```

Comment:
Can we just use a LinkedHashMap for nodesEverSeenById so that the keyset iteration ordering is guaranteed? With this we can then remove nodesEverSeen and perhaps rename nodesEverSeenById to nodesEverSeen.